File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        freq = Counter(nums);
        bucket = [[] for _ in range(len(nums) + 1)]
        for n,f in freq.items():
            bucket[f].append(n)
        res = []
        for i in range(len(bucket)-1, -1, -1):
            for n in bucket[i]:
                res.append(n)
                if len(res) == k:
                    return res

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
    int findMin(vector<int> &nums) {
        int i = 0, d = nums.size() - 1, m;
        while ( i <= d ){
            m = i + ( d - i ) / 2;
            if ( nums[m] < nums[i] )        d = m;
            else if ( nums[m] > nums[d] )   i = m + 1;
            else return nums[i];
        }
    }
};
